docs/examples/code/working_with_signed_urls.md

```diff
@@ -0,0 +1,135 @@
+
+**ref**: https://github.com/developmentseed/titiler/discussions/331
+
+**Goal**: allow users to pass signed url
+
+**requirements**: titiler.core
+
+## What / Why / How
+
+Passing a signed URL for a dataset is not supported by default in TiTiler because the parameters from the signed url are conflicting with the query parameters from the application itself. In order to allow signed url in the application there are two solutions:
+
+
+### 1. URL Encoding
+
+The easiest way (from the application POV) to allow complex URL, is to allow encoded url as input parameter.
+
+```python
+import base64
+url = "http://my.dataset.com/cog.tif?p=aaa&c&1234&t=4321"
+
+# base64.b64encode(url.encode())
+>>> "aHR0cDovL215LmRhdGFzZXQuY29tL2NvZy50aWY/cD1hYWEmYyYxMjM0JnQ9NDMyMQ=="
+```
+
+When base64 encoded, url is just a regular string and thus will be valid in the application.
+
+```python
+"""Minimal COG tiler with Signed URL support."""
+
+import base64
+from titiler.core.factory import TilerFactory
+from titiler.core.errors import DEFAULT_STATUS_CODES, add_exception_handlers
+
+from fastapi import FastAPI
+
+
+# Custom Path dependency which can `decode` a base64 url
+def DatasetPathParams(
+    url: str = Query(..., description="Dataset URL"),
+    base64_encoded: bool = Query(None)
+) -> str:
+    """Create dataset path from args"""
+    if base64_encoded:
+        url = base64.b64decode(url).decode()
+    return url
+
+# Another solution is to pass only the signed url path parameters encoded in base64
```

Comment:
Nit: Instead of keeping as a comment, I'd suggest having two separate code blocks with prose around them, and then a third block that uses `DatasetPathParams`

`````md

Create a PathParams dependency to pass to the tiler, two approaches

```py
# first approach
def DatasetPathParams
```

```py
# second approach
def DatasetPathParams
```

`````